### PR TITLE
Gpt1 for sequence classification

### LINKED ISSUE #7623: Implement PyTorch and/or TensorFlow sequence classification architectures for causal language models
# 🚀 Feature request

The architecture `GPT2ForSequenceClassification` was added in #7501 in PyTorch. It would be great to have it in TensorFlow (cf. issues #7622), but it would also be great to have it for other causal models: ~OpenAI GPT~, ~CTRL~ (PR opened @elk-cloner), ~TransfoXL~ (PR opened @spatil6)

Below is a list of items to follow to make sure the integration of such an architecture is complete:

- Implement `XXXForSequenceClassification` in `modeling_xxx.py` or `TFXXXForSequenceClassification` in `modeling_tf_xxx.py
- Test that architecture in `tests/test_modeling_xxx.py` or `tests/test_modeling_tf_xxx.py`
- Add that architecture to `__init__.py` and ` docs/source/model_doc/xxx.rst`.

Taking a look at the code changes in #7501 would be a good start.

A very good first issue to get acquainted with the library and its architectures!

### PR DESCRIPTION
**Pull request by fmcurti**
# What does this PR do?

Adds sequence classification architecture for GPT-1,
Strongly based on modifications made in #7501

Fixes #7623 (issue) (Partially)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dimiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests), 
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to the it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests? 

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors which may be interested in your PR.

@LysandreJik Here is the new PR without that merge problem, let me know if there is anything that should be changed =)

### CODE REVIEW

**Comment by LysandreJik** [comment]
This is great, thanks for working on it! There's something that I had forgotten in the initial GPT-2 implementation, which was to add it to the auto-models. I did it in this PR: https://github.com/huggingface/transformers/pull/7630.

Could you apply the fix to GPT-1 as well, before we merge?

**Comment by fmcurti** [comment]
> This is great, thanks for working on it! There's something that I had forgotten in the initial GPT-2 implementation, which was to add it to the auto-models. I did it in this PR: #7630.
> 
> Could you apply the fix to GPT-1 as well, before we merge?

Done! 😊

File: src/transformers/modeling_openai.py [hunk 1]
```diff
@@ -732,3 +732,115 @@ def forward(
...
+        self.num_labels = config.num_labels
+        self.transformer = OpenAIGPTModel(config)
+        self.score = nn.Linear(config.n_embd, self.num_labels, bias=False)
+
+        self.init_weights()
+
+    @add_start_docstrings_to_callable(OPENAI_GPT_INPUTS_DOCSTRING)
+    @add_code_sample_docstrings(
+        tokenizer_class=_TOKENIZER_FOR_DOC,
+        checkpoint="microsoft/dialogrpt",
```
Comment (LysandreJik) [inline src/transformers/modeling_openai.py:764]: This checkpoint wouldn't be loadable in `OpenAIGPTForSequenceClassification`, I would just leave `openai-gpt`

File: src/transformers/modeling_openai.py [hunk 2]
```diff
@@ -732,3 +732,115 @@ def forward(
...
+    Since it does classification on the last token, it requires to know the position of the last token.
+    If a :obj:`pad_token_id` is defined in the configuration, it finds the last token that is not a padding token
+    in each row. If no :obj:`pad_token_id` is defined, it simply takes the last value in each row of the batch.
+    Since it cannot guess the padding tokens when :obj:`inputs_embeds` are passed instead of :obj:`input_ids`, it
+    does the same (take the last value in each row of the batch).
+    """,
+    OPENAI_GPT_START_DOCSTRING,
+)
+class OpenAIGPTForSequenceClassification(OpenAIGPTPreTrainedModel):
+    authorized_missing_keys = [r"h\.\d+\.attn\.masked_bias", r"lm_head\.weight"]
```
Comment (LysandreJik) [inline src/transformers/modeling_openai.py:751]: I'm not sure this line is needed. If you remove it and load the `openai-gpt` model, do you get some warnings related to these weights?

**Review by LysandreJik — APPROVED** [review]
This is great! Two last comments and then we'll be good to merge. Thanks for working on it!

File: src/transformers/modeling_openai.py (hunk 2 above)
Comment (fmcurti) [reply src/transformers/modeling_openai.py:751]: Nope, the only warning I get is about score, which I guess is expected:

Some weights of OpenAIGPTForSequenceClassification were not initialized from the model checkpoint at /tmp/tmpiii7di_c and are newly initialized: ['score.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.

**Comment by LysandreJik** [comment]
Thanks @fmcurti!

**Pull request merged**